raft/raft.go

```diff
@@ -565,6 +565,12 @@ func (r *raft) Step(m pb.Message) error {
 	case m.Term > r.Term:
 		lead := m.From
 		if m.Type == pb.MsgVote {
+			if r.state == StateFollower && r.checkQuorum && r.electionElapsed < r.electionTimeout {
```

Comment:
I think to correctly implement this, we also need pre-vote support. 

Or the peer with a higher term will keep on sending vote request. Leader cannot make that peer stable since that peer will always reject the append requests from leader since it has higher term.

Pre-vote can help with this issue since a peer will not increase its term unless it knows there is no leader in the recent election timeout.

Another issue would be peer starting. If a peer get restarted, it will reset the election timeout and cannot vote for the a few seconds or so. If the entire cluster is restarted or is bootstrapped for the first time, there will be a few seconds unavailability.